src/Standards/Generic/Tests/CodeAnalysis/AssignmentInConditionUnitTest.php

```diff
@@ -41,46 +41,54 @@ public function getErrorList()
      * The key of the array should represent the line number and the value
      * should represent the number of warnings that should occur on that line.
      *
+     * @param string $testFile The name of the file being tested.
+     *
      * @return array<int, int>
      */
-    public function getWarningList()
+    public function getWarningList($testFile='AssignmentInConditionUnitTest.1.inc')
```

Comment:
```suggestion
    public function getWarningList($testFile='')
```

Reply:
I went ahead and amended the existing commit to implement this change.

I had added the name of the first test case file as the default value as I was using [InlineControlStructureUnitTest.php](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/src/Standards/Generic/Tests/ControlStructures/InlineControlStructureUnitTest.php#L33) as reference. Is there a reason we want the name of the first test case file as the default value in InlineControlStructureUnitTest::getErrorList() but not in AssignmentInConditionUnitTest::getWarningList()?

Reply:
> I had added the name of the first test case file as the default value as I was using [InlineControlStructureUnitTest.php](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/src/Standards/Generic/Tests/ControlStructures/InlineControlStructureUnitTest.php#L33) as reference. Is there a reason we want the name of the first test case file as the default value in InlineControlStructureUnitTest::getErrorList() but not in AssignmentInConditionUnitTest::getWarningList()?

There are actually three reasons:
1. The default value has no value in practice. It is an optional argument which is not [enforced via the abstract functions](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/tests/Standards/AbstractSniffUnitTest.php#L448-L459), but is [passed in all cases](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/186814f868eef4ad30fb27ec631718bdf62d9a72/tests/Standards/AbstractSniffUnitTest.php#L228-L229), so the default is never used in practice.
2. Setting the default value as file 1 also has an assumption implied, while IMO assumptions have no place in a test suite.
3. Maintainability - one less thing to have to keep in sync and to guard against typos ;-)

Does that help ?

To be honest, it wouldn't be a bad idea to change the default value for `$testFile=...` in all PHPCS native test files which have it to `''` for consistency.

Reply:
It's up for grabs, so yes, feel free to submit a PR for that. Please add a link to my comment in the PR description, so it's clear for others where this is coming from.

Reply:
Here is the PR: https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/168